### ISSUE TITLE
Change video quality in progressive download sources

### ISSUE THREAD

**Issue by JulianKniephoff** [polimediaupv/paella-core#342]
Paella 7 lost the ability to change video quality for progessive download sources. This is confirmed by [the docs](https://paellaplayer.upv.es/#/doc/paella_player_7_presentation.md) in the section "Features lost since Paella Player 6.x" (second paragraph there).

This is slowly coming up as an issue for a couple of our clients to the point where it is blocking them from upgrading. Yes, HLS might be the future, but not everyone is there yet, plus many institutions still have large collections of media published as progressive download that they would like to continue to offer without having to republish and/or -encode everything.

Accordingly, we consider this a rather essential feature for the standard Opencast player, and my question is: Are there any plans to reintroduce it? And if not: Would you be opposed to someone from ELAN trying their hand at contributing the feature?

**Comment by snoesberger** [comment]
I totally agree with Julian. At the University of Bern we currently have over 75000 videos in Opencast, encoded in three different resolutions. We are currently using Paella Player 6 and progressive download to play the videos.
In order to provide the same service with three different resolutions, we would have to re-encode each of the 75000+ videos.
We would love to migrate to Paella 7 and also to HLS streaming. But ideally we could offer all the old videos as progressive download (all 3 resolutions) and only the newly uploaded videos as HLS.

**Comment by turro** [comment]
HI Julian and Sascha

At Valencia we have a similar issue with the legacy videos, and we do have a plugin that allows Paella Player 7 to work with progressive downloads.

However, it has several issues, most of them related to changing resolution while playing a video. In the current state, changing the resultion does a full reload of the interface and goes back to the beggining. Also, going to fullscreen may trigger the same effect if there is a resolution change involved.

We didn't found a good solution for those, and because of that we dropped support of progressive downloads in PP7.

Of course we may share the plugin, and you will be welcome to improve it. However, we are not eager to provide long-term support of it.

Maybe we may set up a meeting next week to discuss options

**Comment by JulianKniephoff** [comment]
Meeting sounds good. 😄 I'll just email you with a few suggestions for an appointment if that's ok with you.

**Comment by snoesberger** [comment]
Unfortunately, I'll be on holiday next week. But of course I'll be interested in the outcome of your discussion.

**Comment by ferserc1** [comment]
Dynamic quality change in progressive download videos was abandoned in paella-core because of the problems it caused. In the current context it is even impossible to implement on some platforms, due to the restrictions imposed by browsers with autoplay videos.

To make this resolution change, the video source must be replaced, playback must start and jump to the previous time instant. There are several actions that must be done asynchronously, and although the main action is triggered by the user, after the first asynchronous response it is usual that the browser blocks the action.

This is detailed in the presentation document:

https://paellaplayer.upv.es/#/doc/paella_player_7_presentation.md

Because of all this, this feature will never be officially implemented, but remember that in case you don't mind dealing with all these problems, you can implement your own video plugin:

https://paellaplayer.upv.es/#/doc/video_plugin.md
